src/main/java/net/sf/jabref/logic/groups/KeywordGroup.java

```diff
@@ -197,17 +200,41 @@ public boolean equals(Object o) {
 
     @Override
     public boolean contains(BibEntry entry) {
-        if (!entry.hasField(searchField)) {
-            return false;
-        }
-        String content = entry.getField(searchField);
         if (regExp) {
+            String content = entry.getField(searchField);
+            if (content == null) {
+                return false;
+            }
             return pattern.matcher(content).find();
         }
+
+        Set<String> words = entry.getFieldAsWords(searchField);
+        if (words.isEmpty()) {
+            return false;
+        }
+
         if (caseSensitive) {
-            return KeywordGroup.containsWord(searchExpression, content);
+            return words.containsAll(searchWords);
+        }
+        return containsCaseInsensitive(searchWords, words);
+    }
+
+    private boolean containsCaseInsensitive(List<String> searchText, Set<String> words) {
+        for (String searchWord : searchText) {
+            if (!containsCaseInsensitive(searchWord, words)) {
```

Comment:
Since this a rather performance-critical method, I decided to use the good old collections interface instead of having the stream overhead.